### PR TITLE
feat: allows dynamic url on useCustomFetch

### PR DESCRIPTION
**Pull request by maximepvrt**

### CODE REVIEW

**Comment by maximepvrt** [comment]
👋 @danielroe

**Comment by maximepvrt** [comment]
@Atinux 🙏

**Review by danielroe — COMMENTED** [review]
What is the issue with the current example?

**Comment by maximepvrt** [comment]
@danielroe 2 points : 
- allows dynamic urls on `useCustomFetch`
- fix cache request by removing key attribute to use default key attribute : the default value is much more comprehensive and includes the baseUrl, parameters, and request method (redefined by my last fix https://github.com/nuxt/nuxt/pull/23086)

**Comment by danielroe** [comment]
I think this is just meant to show an example of a custom fetch composable, and show how to override the defaults.

**Comment by maximepvrt** [comment]
@danielroe The first point may not be critical, but the second one was a significant roadblock for me as it caused unexpected synchronization issues due to edit the key generation strategy. I believe most users won't need to modify the default key strategy even for useCustomFetch, so it could be commented or removed out for simplicity.

File: advanced/use-custom-fetch-composable/composables/useCustomFetch.ts [hunk 1]
```diff
@@ -1,14 +1,12 @@
 import type { UseFetchOptions } from 'nuxt/app'
 import { defu } from 'defu'
 
-export function useCustomFetch<T> (url: string, options: UseFetchOptions<T> = {}) {
+export function useCustomFetch<T> (url: string | (() => string), options: UseFetchOptions<T> = {}) {
```
Comment (danielroe) [inline advanced/use-custom-fetch-composable/composables/useCustomFetch.ts:4]:
```suggestion
export function useCustomFetch<T> (url: string), options: UseFetchOptions<T> = {}) {
```

File: advanced/use-custom-fetch-composable/composables/useCustomFetch.ts [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
   const userAuth = useCookie('token')
   const config = useRuntimeConfig()
 
   const defaults: UseFetchOptions<T> = {
     baseURL: config.baseUrl ?? 'https://api.nuxtjs.dev',
```
Comment (danielroe) [inline advanced/use-custom-fetch-composable/composables/useCustomFetch.ts:9]:
```suggestion
    baseURL: config.baseUrl ?? 'https://api.nuxtjs.dev',
    // this overrides the default key generation, which includes a hash of
    // url, method, headers, etc. - this should be used with care
    key: url,
```

**Review by danielroe — APPROVED** [review]

**Pull request merged**